src/components/v5/common/ActionSidebar/partials/MultiSigSidebar/partials/MultiSigWidget/MultiSigWidget.tsx

```diff
@@ -0,0 +1,93 @@
+import { type FC } from 'react';
+import React from 'react';
+
+import { useAppContext } from '~context/AppContext/AppContext.ts';
+import {
+  type ColonyActionType,
+  type MultiSigUserSignatureFragment,
+  type ColonyMultiSigFragment,
+} from '~gql';
+import { useDomainThreshold } from '~hooks/multiSig/useDomainThreshold.ts';
+import { notMaybe } from '~utils/arrays/index.ts';
+import { getMultiSigRequiredRole } from '~utils/multiSig.ts';
+
+import FinalizeButton from '../FinalizeButton/FinalizeButton.tsx';
+import RemoveVoteButton from '../RemoveVoteButton/RemoveVoteButton.tsx';
+import Signees from '../Signees/Signees.tsx';
+import VoteButton from '../VoteButton/VoteButton.tsx';
+
+const displayName =
+  'v5.common.ActionSidebar.partials.MultiSig.partials.MultiSigWidget';
+
+interface MultiSigWidgetProps {
+  actionType: ColonyActionType;
+  multiSigData: ColonyMultiSigFragment;
+}
+
+const MultiSigWidget: FC<MultiSigWidgetProps> = ({
+  multiSigData,
+  actionType,
+}) => {
+  const { user } = useAppContext();
+  const requiredRole = getMultiSigRequiredRole(actionType);
+
+  const { isLoading, threshold } = useDomainThreshold({
+    domainId: multiSigData.multiSigDomainId,
+    requiredRole,
+  });
+
+  if (isLoading) {
+    return <div>Loading threshold</div>;
+  }
+
+  if (threshold === null) {
+    console.warn('Invalid threshold');
+    return <div>Invalid threshold, assign some stuff</div>;
+  }
+
+  const signatures = (multiSigData?.signatures?.items ?? []).filter(notMaybe);
+  const userSignature = signatures.find(
+    (signature) => signature?.userAddress === user?.walletAddress,
+  );
+
+  const isMultiSigFinalizable = signatures.length >= threshold;
+  const isMultiSigExecuted = multiSigData.isExecuted;
+  const isMultiSigExecutedSuccessfully = multiSigData.isSuccess;
+
+  if (isMultiSigExecuted) {
+    return (
+      <div>
+        MultiSig motion completed{' '}
+        {isMultiSigExecutedSuccessfully ? 'successfully' : 'unsuccessfully'}
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <span>Threshold: {threshold}</span>
+      <Signees
+        signees={signatures as unknown as MultiSigUserSignatureFragment[]}
```

Comment:
this needs typing as `MultiSigUserSignatureFragment` instead of the full type to remove casting